packages/flutter/test/material/bottom_sheet_theme_test.dart

```diff
@@ -0,0 +1,174 @@
+
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        key: scaffoldKey,
+        body: const Center(child: Text('body')),
+      ),
+    ));
+
+    showModalBottomSheet<void>(
+      context: scaffoldKey.currentContext,
+      builder: (BuildContext context) {
+        return Container(
+          child: const Text('BottomSheet'),
+        );
+      },
+    );
+
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+
+    final Material material = _getBottomSheetMaterial(tester);
+    expect(material.color, null);
+    expect(material.elevation, 0.0);
+    expect(material.shape, null);
+  });
+
+  testWidgets('BottomSheet uses values from BottomSheetThemeData', (WidgetTester tester) async {
+    final GlobalKey<ScaffoldState> scaffoldKey = GlobalKey<ScaffoldState>();
+    final BottomSheetThemeData bottomSheetTheme = _bottomSheetTheme();
+
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(bottomSheetTheme: bottomSheetTheme),
+      home: Scaffold(
+        key: scaffoldKey,
+        body: const Center(child: Text('body')),
+      ),
+    ));
+
+    showModalBottomSheet<void>(
+      context: scaffoldKey.currentContext,
+      builder: (BuildContext context) {
+        return Container(
+          child: const Text('BottomSheet'),
+        );
+      },
+    );
+
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+
+    final Material material = _getBottomSheetMaterial(tester);
+    expect(material.color, bottomSheetTheme.backgroundColor);
+    expect(material.elevation, bottomSheetTheme.elevation);
+    expect(material.shape, bottomSheetTheme.shape);
+  });
+
+  testWidgets('BottomSheet widget properties take priority over theme', (WidgetTester tester) async {
+    final GlobalKey<ScaffoldState> scaffoldKey = GlobalKey<ScaffoldState>();
+    const Color backgroundColor = Colors.purple;
+    const double elevation = 7.0;
+    const ShapeBorder shape = RoundedRectangleBorder(
+      borderRadius: BorderRadius.all(Radius.circular(9.0)),
+    );
+
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(bottomSheetTheme: _bottomSheetTheme()),
+      home: Scaffold(
+        key: scaffoldKey,
+        body: const Center(child: Text('body')),
+      ),
+    ));
+
+    showModalBottomSheet<void>(
+      backgroundColor: backgroundColor,
+      elevation: elevation,
+      shape: shape,
+      context: scaffoldKey.currentContext,
+      builder: (BuildContext context) {
+        return Container(
+          child: const Text('BottomSheet'),
+        );
+      },
+    );
+
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+
+    final Material material = _getBottomSheetMaterial(tester);
+    expect(material.color, backgroundColor);
+    expect(material.elevation, elevation);
+    expect(material.shape, shape);
+  });
+}
+
+BottomSheetThemeData _bottomSheetTheme() {
```

Comment:
Just defining this inline makes the tests a tiny bit easier to read.

Reply:
Done